src/libraries/System.Private.Xml/src/System/Xml/Core/XmlWellFormedWriterHelpers.cs

```diff
@@ -403,10 +403,11 @@ internal void Trim()
 
                 // trim the string in StringBuilder
                 string valBefore = _stringValue.ToString();
-                string valAfter = XmlConvert.TrimString(valBefore);
+                ReadOnlySpan<char> valAfter = valBefore.AsSpan().Trim(XmlConvert.WhitespaceChars);
                 if (valBefore != valAfter)
                 {
-                    _stringValue = new StringBuilder(valAfter);
+                    _stringValue = new StringBuilder();
```

Comment:
consider passing in capacity to the constructor since value we append is known

Reply:
valAfter is now ReadOnlySpan<char> and there is no ctor with span 
@krwq